### DISCUSSION TITLE
Search string with refinable doesnt show site pages

### DISCUSSION THREAD

**Discussion by rooobeert** [microsoft-search/pnp-modern-search#3567] [General]
Hi all, 

i got a document library and the site pages library. Both have a site column based on a lookup value of the title of site pages. 
We use this to dynamically show related documents and other site pages on a specific site page. That lookup column is associated with RefinableString05 in the site collections search schema.

The search query for documents is this:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/87785803/3f8de747-27a1-4ede-b09f-6b9ec065bec9)

This shows all documents with the site page's title as a lookup value.

The search query for pages is this:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/87785803/73580695-2013-450e-bc8c-813bd32b8337)

To me this identical except for the path obviously and yet, the one for site pages doesnt give any results.
I have checked if the RefinableString05 is actually filled by displaying it along side the rest of some metadata.
I have also checked that the Parameter Page.Title willl properly translate.

I mean it does show the correct documents. What am I missing?
Is this a bug?

**Comment by wobba** [comment]
Try `.../SitePages/`. Without the *. `path` is special and does prefix matching by default.

**Comment by rooobeert** [comment]
No luck here. 
I tried this: 
path:"https://SPO.sharepoint.com/sites/SITE/SitePages/"
RefinableString05:"{Page.Title}"

**Reply by wobba** [reply]
Question, is it the home page not getting the value? If it's some other page, then try re-index the sitepages library and wait to see if that helps.

**Reply by rooobeert** [reply]
no, we use our own site page content type. We have tried reindexing many times, but as soon as we see that the value is available the filter doesnt work properly and doesnt show a page.

The documents library on the same site collection uses the same site column and refinable and doesnt have any issues with spaces, whatsoever.

**Reply by wobba** [reply]
Which data provider do you use? And I would start simple. Repro the query in e.g. SP Editor browser extension or in Graph Explorer. No query variables, just input. Then add to the KQL and see if you still get results. If all is working, then start looking at the JSON requests being made. Do they look the same as what you manually tested etc.

**Reply by rooobeert** [reply]
I'm using SharePoint. Im verifying a few things on my end now, because it seems like the index is broken and not updating accordingly. I will get back to you latest 2 days.

**Comment by rooobeert** [comment]
@wobba It seems like there was an issue with the index, but that should be resolved by now. 

I still have the issue of whenever a page has a title with spaces, the pnp search result doesnt find the page.

**Reply by wobba** [reply]
If you query for a page with some other means, what does the title look like in RefinableString05? Also, on the request being sent, how is a title with a space encoded or not with the query variable {Page.Title}?

**Reply by rooobeert** [reply]
RefinableString05 is mapped with a lookup column, which should hold both Title and the ID. But the RefinableString05 only holds the Title of the page. 

I was able to query something like RefinableString05:"Name*", which is only part of the title and not from a token.

**Reply by wobba** [reply]
As you didn't reply what I asked I tested and page title with spaces seems to be handled correctly. I used a template of `Title:" {Page.Title}"` which return the expected result at least. If `RefinableString05` contains "This is my page" and you query for `RefinableString05:"This is my page"` I would assume you get a match?

**Comment by rooobeert** [comment]
I feel like SharePoint Online is messing with me. It seems it's related to issues regarding the SharePoint Online search index. It works now but  won't work in two weeks.

In my particular case, the filter now works as expected. I tried the templates you mentioned above, and the results are correct for now.

**Comment by rooobeert** [comment]
Aaaaand here we go again: 
In this I have a results webpart looking for part string of the value that is stored in RefinableString05. In the results in the column test, I have displayed the RefinableString05 value for the documents listed there.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/87785803/0db1d953-d56f-4b16-b265-df02beb9b82d)


If I add the complete string that is stored in the refinable, I get no results: 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/87785803/19aac027-1afa-489d-9777-32c0f603cf51)


And this causes the query with dynamic site title input to also not work properly.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/87785803/19bfde64-a070-4209-babc-e2b2bd67f0d9)


@wobba what can I do?

**Reply by wobba** [reply]
Special characters are always tricky with tokenization. You might want to use RefinableStringWbOffFirst00 instead which has complete matching enabled and see if that works. And also use = equals instead of : contains.

**Reply by rooobeert** [reply]
Alright, I tried both RefinableStringWbOffFirst00 and =, and it seems to be working now.
Does the First00 mean, that if the mapped column is multi selection, it will always only take the first value of the multiple values?

**Reply by wobba** [reply]
First will take the first cp with a value yes, and not all.

**Reply by rooobeert** [reply]
Alright understood. It takes the first value that is not empty. This would be useful if I mapped multiple site columns to one refinablestring. 
I just confirmed that if I map a multi select lookup column to RefinableStringWbOffFirst00, my use case still works. Thanks!
